rs/ic_os/network/mac_address/src/mac_address.rs

```diff
@@ -140,6 +140,24 @@ pub fn get_ipmi_mac() -> Result<FormattedMacAddress> {
     })
 }
 
+/// Get the management MAC address.
+///
+/// Retrieves the MAC address from IPMI if available, else falls back to
+/// a known hard-coded string that is hexadecimally-encoded "invirt".
+pub fn get_mgmt_mac() -> Result<FormattedMacAddress> {
+    let stdout = Command::new("systemd-detect-virt")
+        .output()?
+        .stdout
+        .into_iter()
+        .collect::<Vec<u8>>();
+    let output = std::str::from_utf8(&stdout).unwrap_or("[INVALID UTF8]");
+    match output {
+        "none\n" => get_ipmi_mac(),
+        // The following string is hex for "invirt".
+        _ => FormattedMacAddress::try_from("69:6e:76:69:72:74"),
+    }
```

Comment:
```suggestion
    let output = Command::new("systemd-detect-virt")
        .output()?
        .stdout;

    let stdout = std::str::from_utf8(&output)?.trim();
    
    match stdout {
        "none" => {
             println!("Running on bare metal. Retrieving IPMI MAC address.");
             get_ipmi_mac()
        },
        _ => {
             println!("Virtualization detected. Using hard-coded mac address.");
             FormattedMacAddress::try_from("69:6e:76:69:72:74")
        },
```

Also, gpt told me that:

> Rust crates like sysinfo or heim that provide system information, including virtualization detection.
> 
> use sysinfo::{System, SystemExt};
> 
> fn detect_virtualization() -> Result<Virtualization> {
>     let mut sys = System::new_all();
>     sys.refresh_system();
>     if sys.is_virtual_machine() {
>         ...
>     } else {
>         ...
>     }
> }
> 

I haven't explored this myself though

Reply:
I think it's OK if we avoid another crate.